src/EnergyPlus/UnitarySystem.cc

```diff
@@ -12864,25 +12864,29 @@ namespace UnitarySystems {
                     bool const singleMode = (this->m_SingleMode == 1);
+                    PartLoadFrac = 1.0;
                     for (int speedNum = this->m_CoolingSpeedNum; speedNum <= this->m_NumOfSpeedCooling; speedNum++) {
                         this->m_CoolingSpeedNum = speedNum;
                         state.dataCoilCooingDX->coilCoolingDXs[this->m_CoolingCoilIndex].simulate(
                             state, OperationMode, PartLoadFrac, this->m_CoolingSpeedNum, this->m_CoolingSpeedRatio, this->m_FanOpMode, singleMode);
                         if ((state.dataLoopNodes->Node(OutletNode).Temp - DesOutTemp) < Acc) break;
                     }
 
-                    std::array<Real64, 8> Par2 = {double(this->m_CoolingCoilIndex),
-                                                  DesOutTemp,
-                                                  // dehumidification mode = 0 for normal mode, 1+ for enhanced mode
-                                                  // need to test what happens when Alt mode doesn't exist, or somehow test for it,
-                                                  // or fatal out in GetInput
-                                                  1.0, // DehumidMode
-                                                  double(FanOpMode),
-                                                  double(this->m_CoolingSpeedNum),
-                                                  1.0, //  this->m_CoolingSpeedRatio;
-                                                  0.0,
-                                                  0.0};
-                    General::SolveRoot(state, Acc, MaxIte, SolFla, PartLoadFrac, &this->genericDXCoilResidual, 0.0, 1.0, Par2);
+                    // make sure outlet temp is below set point before calling SolveRoot
+                    if ((state.dataLoopNodes->Node(OutletNode).Temp - DesOutTemp) < Acc) {
+                        std::array<Real64, 8> Par2 = {double(this->m_CoolingCoilIndex),
+                                                      DesOutTemp,
+                                                      // dehumidification mode = 0 for normal mode, 1+ for enhanced mode
+                                                      // need to test what happens when Alt mode doesn't exist, or somehow test for it,
+                                                      // or fatal out in GetInput
+                                                      1.0, // DehumidMode
+                                                      double(FanOpMode),
+                                                      double(this->m_CoolingSpeedNum),
+                                                      1.0, //  this->m_CoolingSpeedRatio;
+                                                      0.0,
+                                                      0.0};
+                        General::SolveRoot(state, Acc, MaxIte, SolFla, PartLoadFrac, &this->genericDXCoilResidual, 0.0, 1.0, Par2);
+                    }
```

Comment:
Final revisions: protect SolveRoot with check on bounds. Note the new if test here is identical to the break in the for loop above at line 12872. If this new if test failes it means the for loop above went to max speed without reaching the set point and SolveRoot would have failed with -2. If this new if test passes it means that speed n could reach the set point and SolveRoot can now iterate on a solution at speed n. Why these subtle changes have not shown up before probably means we have no example file or unit test coverage.